tests/golos.publication_tests.cpp

```diff
@@ -0,0 +1,439 @@
+                           ("voter", voter)
+                           ("author", author)
+                           ("permlink", permlink)
+                           ("weight", weight)
+        );
+    }
+
+    action_result create_battery(account_name account) {
+        return push_action(account, N(createacc), mvo()
+                           ("name", account)
+                          );
+    }
+
+    void require_equal(const fc::variant &obj1, const fc::variant &obj2) {
+        BOOST_REQUIRE_EQUAL(true, obj1.is_object() && obj2.is_object());
+        BOOST_REQUIRE_EQUAL(obj1.get_object()["id"].as<uint64_t>(), obj2.get_object()["id"].as<uint64_t>());
+        BOOST_REQUIRE_EQUAL(obj1.get_object()["account"].as<account_name>(), obj2.get_object()["account"].as<account_name>());
+//        BOOST_REQUIRE_EQUAL(obj1.get_object()["permlink"].as<checksum_type>(), obj2.get_object()["permlink"].as<checksum_type>());
+        BOOST_REQUIRE_EQUAL(obj1.get_object()["parentacc"].as<account_name>(), obj2.get_object()["parentacc"].as<account_name>());
+        BOOST_REQUIRE_EQUAL(obj1.get_object()["parentprmlnk"].as<std::string>(), obj2.get_object()["parentprmlnk"].as<std::string>());
+        BOOST_REQUIRE_EQUAL(obj1.get_object()["curatorprcnt"].as<uint64_t>(), obj2.get_object()["curatorprcnt"].as<uint64_t>());
+        BOOST_REQUIRE_EQUAL(obj1.get_object()["payouttype"].as<std::string>(), obj2.get_object()["payouttype"].as<std::string>());
+        BOOST_REQUIRE_EQUAL(obj1.get_object()["paytype"].as<std::string>(), obj2.get_object()["paytype"].as<std::string>());
+    }
+
+    void require_equal_battery(const fc::variant &obj1, const fc::variant &obj2) {
+        BOOST_REQUIRE_EQUAL(true, obj1.is_object() && obj2.is_object());
+        BOOST_REQUIRE_EQUAL(obj1.get_object()["posting_battery"].get_object()["charge"].as<uint16_t>(),        obj2.get_object()["posting_battery"].get_object()["charge"].as<uint16_t>());
+        BOOST_REQUIRE_EQUAL(obj1.get_object()["battery_of_votes"].get_object()["charge"].as<uint16_t>(),       obj2.get_object()["battery_of_votes"].get_object()["charge"].as<uint16_t>());
+        BOOST_REQUIRE_EQUAL(obj1.get_object()["limit_battery_posting"].get_object()["charge"].as<uint16_t>(),  obj2.get_object()["limit_battery_posting"].get_object()["charge"].as<uint16_t>());
+        BOOST_REQUIRE_EQUAL(obj1.get_object()["limit_battery_comment"].get_object()["charge"].as<uint16_t>(),  obj2.get_object()["limit_battery_comment"].get_object()["charge"].as<uint16_t>());
+        BOOST_REQUIRE_EQUAL(obj1.get_object()["limit_battery_of_votes"].get_object()["charge"].as<uint16_t>(), obj2.get_object()["limit_battery_of_votes"].get_object()["charge"].as<uint16_t>());
+    }
+
+    abi_serializer abi_ser;
+};
+
+BOOST_AUTO_TEST_SUITE(golos_publication_tests)
+
+BOOST_FIXTURE_TEST_CASE(create_post, golos_publication_tester) try {
+    BOOST_REQUIRE_EQUAL(success(), create_battery(N(golos.pub)));
```

Comment:
Please, use `BOOST_CHECK_…` instead of `BOOST_REQUIRE_…`. Decided to use it in Golos code